src/extension/tags.c

```diff
@@ -919,6 +1075,30 @@ static bool _set_appsec_enabled(zval *metrics_zv)
            NULL;
 }
 
+bool dd_parse_automated_user_events_tracking(
+    zai_string_view value, zval *nonnull decoded_value, bool persistent)
+{
+    if (!value.len) {
+        return false;
+    }
+
+    bool result = false;
+    size_t len = strlen((const char *)value.ptr);
+    if (dd_string_equals_lc(value.ptr, len, ZEND_STRL(MODE_SAFE)) ||
+        dd_string_equals_lc(value.ptr, len, ZEND_STRL(MODE_EXTENDED)) ||
+        dd_string_equals_lc(value.ptr, len, ZEND_STRL(MODE_DISABLED))) {
+        result = true;
+    }
+
+    if (result) {
+        ZVAL_STR(decoded_value, zend_string_alloc(value.len, persistent));
```

Comment:
Would it not be possible to convert the decoded value to some enum instead?

Reply:
Do you mean having an additional global variable? If we make this configuration a int, on the phpinfo customers will see numbers instead of the value they set

Reply:
Sure, a "global" thread-local variable could work.

Reply:
done